src/processing.rs

```diff
@@ -0,0 +1,61 @@
+//! Everything related to the markdown processing
+use glob::glob;
+use markdown::to_html;
+
+use std::fs::File;
+use std::path::PathBuf;
+use std::io::prelude::*;
+use std::str;
+
+#[derive(Default)]
+/// Global processing structure
+pub struct Processing {
+    /// A collection of paths for the processing
+    paths: Vec<PathBuf>,
+}
+
+impl Processing {
+    /// Reads all markdown files recursively from a given directory
+    /// Clears the current available paths
+    pub fn read_from_directory(&mut self, directory: &str) {
+        /// Remove all paths
+        self.paths.clear();
+
+        /// Gather new content
+        let md_path = PathBuf::from(&directory).join("**/*.md");
```

Comment:
I would prefer .join(**).join(*.md) becaus not every OS works with "/" as filesystem seperator.